### ISSUE TITLE
Parsing error: Unexpected token

### ISSUE THREAD

**Issue by BridgeAR** [eslint/eslint#10555]
**Tell us about your environment**

* **ESLint Version:** 5.0.0
* **Node Version:** 10.5.0
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

See the Node.js configuration.

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
console.log(`abc ${}`);
```

**What did you expect to happen?**

It should report that the template string is faulty. It should not fail.

**What actually happened? Please include the actual, raw output from ESLint.**

![image](https://user-images.githubusercontent.com/8822573/42263708-14fd650a-7f6f-11e8-8196-08fe0362f7f9.png)

**Comment by ljharb** [comment]
That should be a parsing error tho; template literal interpolations can't be empty

**Comment by BridgeAR** [comment]
It is of course "correct" that it is a unexpected token error but to me it feels weird when eslint reports parsing errors. Especially when linting happens in the background.

**Comment by not-an-aardvark** [comment]
Since ESLint provides an AST to all rules, ESLint needs to be able to parse a file in order to run rules on it. If a file contains invalid syntax, ESLint has no choice but to report a parsing error.

**Comment by BridgeAR** [comment]
I know about the reason, I just wondered if it would be possible to report something nicer instead. I am fine with closing this though.
